app/models/solidus_afterpay/payment_method.rb

```diff
@@ -33,24 +31,39 @@ def try_void(payment)
     end
 
     def available_for_order?(order)
-      available_payment_currency == order.currency && available_payment_range.include?(order.total)
+      currency == order.currency && available_payment_range.include?(order.total)
     end
 
     private
 
     def available_payment_range
-      min = preferred_minimum_amount || configuration&.minimumAmount&.amount.to_f
-      max = preferred_maximum_amount || configuration&.maximumAmount&.amount.to_f
+      minimum_amount..maximum_amount
+    end
 
-      min..max
+    def configuration
+      @configuration ||= gateway.retrieve_configuration
     end
 
-    def available_payment_currency
-      preferred_currency || configuration&.maximumAmount&.currency
+    def minimum_amount
+      Rails.cache.fetch("solidus_afterpay_min_amount", expires_in: expires_in_seconds) do
+        configuration&.minimumAmount&.amount.to_f
```

Comment:
We should find a way to handle `configuration` errors. The `retrieve_configuration` method returns `nil` in case of Afterpay errors (e.g. request timeout), and invalid values will be cached for one day. Maybe we can use something like this (untested):
```ruby
Rails.cache.fetch("solidus_afterpay_min_amount", expires_in: expires_in_seconds) do
  configuration&.minimumAmount&.amount.to_f
end.tap do |_|
  Rails.cache.delete('solidus_afterpay_min_amount') if configuration.nil?
end
```

@vassalloandrea WDYT?

Reply:
NVM, this will call `configuration` even if the value is cached. We should find another way to handle it

Reply:
```ruby
if Rails.cache.exist?(:solidus_afterpay_min_amount)
  Rails.cache.read(:solidus_afterpay_min_amount)
else
  Rails.cache.write(
    :solidus_afterpay_min_amount,
    configuration&.minimumAmount&.amount.to_f,
    expires_in: expires_in_seconds
  ) unless configuration.nil?

  configuration&.minimumAmount&.amount.to_f
end
```

Reply:
I know it's ugly but it should work and we can work on it

Reply:
I think that this global `fetch_configuration` method is missing @garciajordy

Reply:
Lol Didn't see that comment!